easybuild/framework/easystack.py

```diff
@@ -108,21 +118,53 @@ class EasyStackParser(object):
 
     @staticmethod
     def parse(filepath):
-        """Parses YAML file and assigns obtained values to SW config instances as well as general config instance"""
+        """
+        Parses YAML file and assigns obtained values to SW config instances as well as general config instance"""
         yaml_txt = read_file(filepath)
 
         try:
             easystack_raw = yaml.safe_load(yaml_txt)
         except yaml.YAMLError as err:
             raise EasyBuildError("Failed to parse %s: %s" % (filepath, err))
 
-        easystack = EasyStack()
+        easystack_data = None
+        top_keys = ('easyconfigs', 'software')
+        for key in top_keys:
+            if key in easystack_raw:
+                easystack_data = easystack_raw[key]
+                break
 
-        try:
-            software = easystack_raw["software"]
-        except KeyError:
-            wrong_structure_file = "Not a valid EasyStack YAML file: no 'software' key found"
-            raise EasyBuildError(wrong_structure_file)
+        if easystack_data is None:
+            msg = "Not a valid EasyStack YAML file: no 'easyconfigs' or 'software' top-level key found"
+            raise EasyBuildError(msg)
+        else:
+            parse_method_name = 'parse_by_' + key
+            parse_method = getattr(EasyStackParser, 'parse_by_%s' % key, None)
+            if parse_method is None:
+                raise EasyBuildError("Easystack parse method '%s' not found!", parse_method_name)
+
+        # assign general easystack attributes
+        easybuild_version = easystack_raw.get('easybuild_version', None)
+        robot = easystack_raw.get('robot', False)
+
+        return parse_method(filepath, easystack_data, easybuild_version=easybuild_version, robot=robot)
```

Comment:
Indeed, I've left that here for now, but we'll definitely get rid of `robot=` going forward when we add proper support for specifying options for specific entries. And we can since support for easystack files is still marked as experimental...